app/middlewares/authenticator.js

```diff
@@ -2,13 +2,19 @@ const logger = require('../logger');
 const { decodeToken } = require('../helpers/users');
 const { authenticationError } = require('../errors');
 const { findUserByEmail } = require('../services/users');
-const { TOKEN_INVALID } = require('../../config/constants/errorMessages');
+const { TOKEN_INVALID, USER_NOT_ALLOWED } = require('../../config/constants/errorMessages');
+const { ROLE } = require('../../config/constants/users_constants');
+const { ADMIN_USERS } = require('../../config/constants/routes');
 
 exports.authenticator = async (req, res, next) => {
   try {
     const tokenVerified = decodeToken(req.headers.token);
     logger.info('Token is valid');
     const userFound = await findUserByEmail(tokenVerified.mail);
+    if (userFound.length && req.url === ADMIN_USERS && userFound[0].dataValues.role !== ROLE.admin) {
```

Comment:
Do not do the routes comparison here, but in the routes file. You can "clone" the authenticator method adding the admin check, or even better, add a new method just to validate if admin or not and passing that middleware in the routes file to the required endpoints.

Reply:
Also, it is better to return a 403 error for authorization, instead of a 401. (401 for authentication, aka wrong pass, and 403 for lack of permission, aka not the right role)

Reply:
Yes, you are rigth.